File: clients/java/src/main/java/model/BulletParams.java
package model;

import util.StreamUtil;

public class BulletParams {
    private double speed;
    public double getSpeed() { return speed; }
    public void setSpeed(double speed) { this.speed = speed; }
    private double size;
    public double getSize() { return size; }
    public void setSize(double size) { this.size = size; }
    private int damage;
    public int getDamage() { return damage; }
    public void setDamage(int damage) { this.damage = damage; }
    public BulletParams() {}
    public BulletParams(double speed, double size, int damage) {
        this.speed = speed;
        this.size = size;
        this.damage = damage;
    }
    public static BulletParams readFrom(java.io.InputStream stream) throws java.io.IOException {
        BulletParams result = new BulletParams();
        result.speed = StreamUtil.readDouble(stream);
        result.size = StreamUtil.readDouble(stream);
        result.damage = StreamUtil.readInt(stream);
        return result;
    }
    public void writeTo(java.io.OutputStream stream) throws java.io.IOException {
        StreamUtil.writeDouble(stream, speed);
        StreamUtil.writeDouble(stream, size);
        StreamUtil.writeInt(stream, damage);
    }
}

File: clients/javascript/model/bullet-params.js
class BulletParams {
    constructor(speed, size, damage) {
        this.speed = speed;
        this.size = size;
        this.damage = damage;
    }
    static async readFrom(stream) {
        let speed;
        speed = await stream.readDouble();
        let size;
        size = await stream.readDouble();
        let damage;
        damage = await stream.readInt();
        return new BulletParams(speed, size, damage);
    }
    async writeTo(stream) {
        let speed = this.speed;
        await stream.writeDouble(speed);
        let size = this.size;
        await stream.writeDouble(size);
        let damage = this.damage;
        await stream.writeInt(damage);
    }
}
module.exports = { BulletParams: BulletParams }
